gcp-auth-extension/README.md

```diff
@@ -36,7 +36,10 @@ Here is a list of configurable options for the extension:
 
 - `GOOGLE_CLOUD_PROJECT`: Environment variable that represents the Google Cloud Project ID to which the telemetry needs to be exported.
   - Can also be configured using `google.cloud.project` system property.
-  - If this option is not configured, the extension would infer GCP Project ID from the application default credentials. For more information on application default credentials, see [here](https://cloud.google.com/docs/authentication/application-default-credentials).
+  - This is a required option, the agent configuration will fail if this option is not set.
+- `GOOGLE_CLOUD_QUOTA_PROJECT`: Environment variable that represents the Google Cloud Quota Project ID which will be charged for the GCP API usage. To learn more about a *quota project*, see [here](https://cloud.google.com/docs/quotas/quota-project).
+  - Can also be configured using `google.cloud.quota.project` system property.
+  - If this option is not configured, the extension would infer GCP Quota Project ID from the application default credentials. For more information on application default credentials, see [here](https://cloud.google.com/docs/authentication/application-default-credentials).
```

Comment:
nit the extension doesn't infer the quota project id right? It just omits it to get the default behavior.

Reply:
If not configured (set by the user), the quota project ID from the application default credentials is used as fallback.

If the inferred/resolved value of the quota project ID is null or empty, the header is omitted.

Reply:
OK I think my confusion is "infer" to me implies that the extension will always generate some kind quota project from ADC (not let it be null). I assumed it would try to set with ADC's project ID and always send

I think something like "it will use quota project found with ADC if available" and mention that this config is not required.

Reply:
I will update the wording to be more clear.